scapy/fields.py

```diff
@@ -460,7 +464,10 @@ def getfield(self, pkt, s):
             l = self.length_from(pkt)
         elif self.count_from is not None:
             c = self.count_from(pkt)
-            
+        elif not isinstance(self.next_cls_cb, type(None)):
```

Comment:
Same remark for the test. Also, why `elif`? This does not apply if `self.length_from` or `self.count_from` is set?